### DISCUSSION TITLE
Could network attribute be embedded within WebRAVE?

### DISCUSSION THREAD

**Discussion by philipbaileynar** [Riverscapes/riverscapes-tools#296] [General]
WARNING: considerable blue sky thinking in what follows!

- We have too many tools!
- Why is network attribute tool a separate web page / app?
- Could network attribute be a feature/button in WebRAVE?

Obviously the way the tool is structured right now it requires the catchment and metrics to be calculated *a priori*... but either:

- we only enable the button in places where network attributes have been calculated (funded)
- we switch gears and calculate on the fly!

FYI @MattReimer @joewheaton

**Comment by MattReimer** [comment]
I won't speak to the ideas here (@joewheaton ) but from a purely technical perspective and as it stands right now I would have to say no. The network attribute tool shares almost nothing in common on the front-end or the back-end with WebRave or riverscapes project in its current iteration. 

* Webrave is a tool to visualize riverscapes projects. Its dimensionality is the bounds of a project and no more. The interactivity is extremely limited. We might end up being able to identify features but that's about it.

* network attribute tool visualizes queries and the contents of a geospatial database. The dimensionality is whatever the size of the upstream catchments are. It is a much more interactive tool with lots of user clicking and zooming around in order to get the catchments in frame. 

Now, if you say you want two tools crammed into the same website that's something we can look into. I'm really not sure how we implement an interface that gets people from one of these ideas to the next though.

That's just my $0.02

**Reply by philipbaileynar** [reply]
Just to be completely clear...

1.  WebRAVE has a button for network attributes
2. The decision to enable/disable the button is driven by a single GeoJSON polygon somewhere. Or maybe not at all. You click somewhere not yet run and you get back no results and WebRAVE says "sorry, contact sys admin to pay for net attributes in this location")
3. User clicks web map.
4. WebRAVE sends coordinates to network attributes API
5. API returns a single JSON object with two items:
    - GeoJSON catchment polygon
    - Dictionary of metrics
6. WebRAVE displays semi-transparent polygon (regardless of what else is in the map) and metrics display similar to how layer metadata does now.

WebRAVE doesn't really need to know anything about network attributes. We don't care if the NHD polylines are displayed or not. This is the beauty of the network attribute API. XY coordinates in and GeoJSON and metrics out!
